### PR TITLE
Add users invite

### PR DESCRIPTION
**Pull request by rwhite27**

### CODE REVIEW

**Comment by rwhite27** [comment]
Basically added users invite featurea and its tests

File: api/controllers/UsersController.php [hunk 1]
```diff
@@ -226,4 +237,94 @@ public function devices(): Response
...
+        $request['created_at'] = date('Y-m-d H:m:s');
+
+        if ($company = Companies::findFirst($this->userData->default_company)) {
+            $request['name'] = $company->name;
+            $request['default_company'] = $this->userData->default_company;
+        }
+
+        //Lets insert the new user to our system.
+
+        if ($this->model->save($request, $this->createFields)) {
```
Comment (kaioken) [inline api/controllers/UsersController.php]: @rwhite27 move all the logic to after save

File: tests/api/UsersInviteCest.php [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+        $I->seeResponseIsSuccessful();
+        $response = $I->grabResponse();
+        $data = json_decode($response, true);
+
+        $I->assertTrue($data['email'] == $testEmail);
+
+        $hash = $data['invite_hash'];
+
+        $I->sendPost('/v1/users/invite/insert?hash=' . $hash, [
```
Comment (kaioken) [inline tests/api/UsersInviteCest.php]: @rwhite27 `/user-invites/{hash}`

File: api/controllers/AuthController.php [hunk 3]
```diff
@@ -77,4 +94,52 @@ protected function sendEmail(BakaUsers $user, string $type): void
...
+            }
+        }
+
+        //Save data to users_invite table and generate a hash for the invite
+        $userInvite = new UsersInvite();
+        $userInvite->company_id = $this->userData->default_company;
+        $userInvite->app_id = $this->app->getId();
+        $userInvite->role_id = $request['role'] == 'Admins' ? 1 : 2;
+        $userInvite->email = $request['email'];
+        $userInvite->invite_hash = hash('md5', $request['email']);
```
Comment (kaioken) [inline api/controllers/AuthController.php]: @rwhite27 usa random / uuid from phalcon

**Pull request merged**